src/bridge/bridge.c

```diff
@@ -173,66 +173,41 @@ send_init_command (CockpitTransport *transport,
   g_bytes_unref (bytes);
 }
 
-static void
-setup_dbus_daemon (gpointer addrfd)
-{
-  g_unsetenv ("G_DEBUG");
-  cockpit_unix_fd_close_all (3, GPOINTER_TO_INT (addrfd));
-}
-
-static GPid
+static GSubprocess *
 start_dbus_daemon (void)
 {
-  GError *error = NULL;
-  GString *address = NULL;
+  g_autoptr(GError) error = NULL;
   gchar *line;
   gsize len;
   gssize ret;
-  GPid pid = 0;
-  gchar *print_address = NULL;
+  GSubprocess *process = NULL;
   int addrfd[2] = { -1, -1 };
-  GSpawnFlags flags;
-
-  gchar *dbus_argv[] = {
-      "dbus-daemon",
-      "--print-address=X",
-      "--session",
-      NULL
-  };
 
   if (pipe (addrfd))
     {
       g_warning ("pipe failed to allocate fds: %m");
-      goto out;
+      return NULL;
     }
 
-  print_address = g_strdup_printf ("--print-address=%d", addrfd[1]);
-  dbus_argv[1] = print_address;
-
   /* The DBus daemon produces useless messages on stderr mixed in */
-  flags = G_SPAWN_LEAVE_DESCRIPTORS_OPEN | G_SPAWN_SEARCH_PATH |
-          G_SPAWN_STDERR_TO_DEV_NULL | G_SPAWN_STDOUT_TO_DEV_NULL;
-
-  g_spawn_async_with_pipes (NULL, dbus_argv, NULL, flags,
-                            setup_dbus_daemon, GINT_TO_POINTER (addrfd[1]),
-                            &pid, NULL, NULL, NULL, &error);
-
-  close (addrfd[1]);
+  g_autoptr(GSubprocessLauncher) dbus_daemon = g_subprocess_launcher_new (G_SUBPROCESS_FLAGS_STDOUT_SILENCE |
+                                                                          G_SUBPROCESS_FLAGS_STDERR_SILENCE);
+  g_subprocess_launcher_unsetenv (dbus_daemon, "G_DEBUG");
+  g_subprocess_launcher_take_fd (dbus_daemon, addrfd[1], 4);
```

Comment:
Why 4?

Reply:
Right, should have taken 42! (I looked at the number in the original code, and it was 4, that must have stuck in  my brain). Changed to 3.